osu.Game/Overlays/Chat/ChatLine.cs

```diff
@@ -112,16 +116,116 @@ public ChatLine(Message message)
                     Padding = new MarginPadding { Left = message_padding + padding },
                     Children = new Drawable[]
                     {
-                        new OsuSpriteText
+                        textContainer = new TextFlowContainer
                         {
-                            Text = Message.Content,
-                            TextSize = text_size,
                             AutoSizeAxes = Axes.Y,
                             RelativeSizeAxes = Axes.X,
                         }
                     }
                 }
             };
+
+            string toParse = message.Content;
+            List<SplitMarker> markers = new List<SplitMarker>();
+            markers.AddRange(parseSplitMarkers(ref toParse, "**", SplitType.Bold));
+            markers.AddRange(parseSplitMarkers(ref toParse, "*", SplitType.Italic));
+            markers.AddRange(parseSplitMarkers(ref toParse, "_", SplitType.Italic));
+
+            // Add a sentinel marker for the end of the string such that the entire string is rendered
+            // without requiring code duplication.
+            markers.Add(new SplitMarker { Index = toParse.Length, Length = 0, Type = SplitType.None });
+
+            // Sort markers from earliest to latest
+            markers.Sort((a, b) => a.Index.CompareTo(b.Index));

```

Comment:
Make `SplitMarker : IComparable<SplitMarker>`, and then default sort method can be used.